reasoners/lm/hf_model.py

```diff
@@ -1 +1,252 @@
+
+        if quantized == "8bit":
+            self.model = LlamaForCausalLM.from_pretrained(
+                model_pth,
+                load_in_8bit=True,
+                torch_dtype=torch.float16,
+                device_map="auto",                                    
+            )
+        elif quantized == "nf4" or quantized  == "fp4":
+            bnb_config = BitsAndBytesConfig(
+                load_in_4bit=True,
+                bnb_4bit_use_double_quant=True,
+                bnb_4bit_quant_type=quantized,
+                bnb_4bit_compute_dtype=torch.bfloat16,
+            )
+
+            print("quantizing.............................")
+
+            self.model = LlamaForCausalLM.from_pretrained(
+                model_pth,
+                quantization_config=bnb_config,
+                device_map="auto",
+            )
+        elif quantized == "awq":
+            config = AutoConfig.from_pretrained(model_pth, trust_remote_code=True)
+            self.tokenizer = AutoTokenizer.from_pretrained(model_pth, trust_remote_code=True, lagacy=False)
+            kwargs = {"torch_dtype": torch.float16, "low_cpu_mem_usage": True}
+            self.model = LlamaForCausalLM.from_pretrained(model_pth, config=config, trust_remote_code=True,**kwargs)
+            self.model.eval()
+            awq_results = torch.load(load_awq_pth, map_location="cpu")
+            apply_awq(self.model, awq_results)
+            q_config = {
+                "zero_point": True,  # by default True
+                "q_group_size": 128,  # whether to use group quantization
+
+            }
+            real_quantize_model_weight(self.model, w_bit=4, q_config=q_config)
+            kwargs = {"max_memory": None}
+            device_map = infer_auto_device_map(self.model,no_split_module_classes=["OPTDecoderLayer", "LlamaDecoderLayer", "BloomBlock", "MPTBlock", "DecoderLayer"], **kwargs)
+            self.model = dispatch_model(self.model, device_map=device_map)
+
+        else:
+            self.model = LlamaForCausalLM.from_pretrained(
+                model_pth,
+                device_map="auto",
+            )
+        if peft_pth is not None:
+            self.model = PeftModel.from_pretrained(
+                self.model, 
+                peft_pth,
+                torch_dtype=torch.float16
+            )
+        
+        self.max_new_tokens = max_new_tokens
+        self.max_batch_size = max_batch_size
+        self.max_length = max_length
+        self.device = device
+        self.model = BetterTransformer.transform(self.model)
+        self.model.eval()
+        # for old llama tokenizer's config, below is necessary
+        self.model.config.pad_token_id = self.tokenizer.pad_token_id = 0  # unk
+        self.model.config.bos_token_id = 1
+        self.model.config.eos_token_id = 2
+        # if torch.__version__ >= "2" and sys.platform != "win32":#need to figure out this line
```

Comment:
I think this is a new feature in pytorch2 to accelerate? If it's not runnable maybe we can simply delete it.

Reply:
Yes, it's temporarily not runnable. See https://github.com/pytorch/pytorch/issues/107269.